screens/HomeScreen.tsx

```diff
@@ -44,73 +40,124 @@ export default function HomeScreen({
+              );
+            }}
           />
+          <View style={styles.content}>
+            <Notification
+              title='Nieuwe stage Plek'
+              message='Er is heeft zich een nieuwe stage plek bij jou in de buurt aangemeld.'
+            />
+            <Notification
+              title='Tip !'
+              message='Voeg een foto toe aan je profiel om meer kans te krijgen op een stage plek.'
+            />
+            <Notification
+              title='Je hebt een match'
+              message='Er is heeft zich een nieuwe stage plek bij jou in de buurt aangemeld'
+            />
+          </View>
+          <NavBar />
         </View>
-        <NavBar />
-      </View>
-    );
+      );
+    case 'Moderator':
+      if (!companies) {
+        backendFetch<User[]>('GET', 'company')
+          .then(companies => setCompanies(companies as User[]))
+          .catch(alert);
+      }
 
-  return (
-    <View style={styles.main}>
-      <TopBar
-        ScreenName={'Companies'}
-        Press={async () => {
-          await SecureStore.deleteItemAsync('login-token').then(r =>
-            navigate('WelcomeScreen')
-          );
-        }}
-      />
+      return (
+        <View style={styles.main}>
+          <TopBar
+            ScreenName={'Companies'}
+            Press={async () => {
+              await SecureStore.deleteItemAsync('login-token').then(() =>
+                navigate('WelcomeScreen')
+              );
+            }}
+          />
 
-      <ScrollView style={styles.content}>
-        {companies?.map(e => (
-          <Notification
-            title={e.name}
-            message={e.email + (e.approved ? ' (approved)' : ' (unapproved)')}
-            key={e.name}
-            action={async o => {
-              backendFetch('POST', 'company/approve/' + e.id)
-                .then(o => {
-                  e.approved = !e.approved;
-                  setCompanies([...companies]);
-                })
-                .catch(alert);
+          <ScrollView style={styles.content}>
+            {companies?.map(company => (
+              <Notification
+                title={company.name}
+                message={
+                  company.email +
+                  (company.approved ? ' (approved)' : ' (unapproved)')
+                }
+                key={company.name}
+                action={async () => {
+                  backendFetch('POST', 'company/approve/' + company.id)
+                    .then(() => {
+                      company.approved = !company.approved;
+                      setCompanies([...companies]);
+                    })
+                    .catch(alert);
+                }}
+                buttonMessage={company.approved ? 'unapprove' : 'approve'}
+              />
+            ))}
+          </ScrollView>
+        </View>
+      );
+    case 'Company':
+      if (!flirts) {
+        backendFetch<Flirt[]>('GET', 'flirts').then(flirts =>
```

Comment:
Fetches should *always* be in a function that gets called *once*. Such as useEffect or an onClick handler